src/ol/featureloader.js

```diff
@@ -87,8 +87,10 @@ ol.featureloader.tile = function(url, format) {
        * @this {ol.VectorTile}
        */
       function(features, dataProjection) {
-        this.setProjection(dataProjection);
-        this.setFeatures(features);
+        /** @type {ol.VectorTile} */
+        var me = this;
```

Comment:
OK, I got it working by add a dependency:
```
goog.require('ol.VectorTile');
```.
It is a bit of a pity though since it forces every one using the feature loarder to embed the code for vector tile. I think the solution from @sbrunner is preferable in this case.